src/Front/Pattern/Anomaly.hs

```diff
@@ -119,6 +119,7 @@ checkExpr (A.Match e arms) =
   let (ps, es) = unzip arms in checkExpr e >> checkExprs es >> checkPats ps
 checkExpr (A.CQuote _  ) = return ()
 checkExpr (A.CCall _ es) = mapM_ checkExpr es
+checkExpr (A.ListExpr _)    = return ()
```

Comment:
Instead of returning unit here, could you recurse on the elements of the list? Replace the wildcard with a name and map checkExpr over them?